### ISSUE TITLE
js_unit_tests and lint_watch_without_args_tests are flaky

### ISSUE THREAD

**Issue by WenheLI** [denoland/deno#13019]
When I was running deno's source code on master branch, I found some test cases will fail even though no code has been changed.
```
failures:
    integration::js_unit_tests
    integration::watcher::lint_watch_without_args_test
failures:

        netTcpListenIteratorBreakClosesResource
```

See the above list. Is this something related to the test cases or my dev env?

**Comment by kt3k** [comment]
Does this error persist? What OS do you use?

**Comment by WenheLI** [comment]
Mine OS is MacOS. And the error is there for most of the time.

**Comment by kt3k** [comment]
hmm we don't observe flakiness on those test cases in CI. So I think it's specific to your environment though I don't see the reason at the moment..

**Comment by kitsonk** [comment]
I'm on MacOS and run the CI locally quite often and haven't noticed those failures.

**Comment by WenheLI** [comment]
Hummm. I think it is related with #13035 .
I will run into error when trigger `integration::watcher::lint_watch_without_args_test`.
And this will be resolved when I run cargo clean.

**Comment by dsherret** [comment]
@WenheLI yeah, it happens to me almost every time when I run the test on its own in WSL. I think it's because our debouncer doesn't actually debounce (or I'm misunderstanding it... code seems overly complex), which would mitigate the issue if it did. I will open a PR soon.

**Comment by dsherret** [comment]
Let's close this now that #13038 is merged and #13040 is about to be. It should be better now.
